### PR TITLE
Enclave tls: add sgx local report instance

### PR DESCRIPTION
**Pull request by YangLiang3**

### CODE REVIEW

File: enclave-tls/src/Makefile [hunk 1]
```diff
@@ -27,7 +27,8 @@ enclave_quotes := $(addsuffix .so,$(addprefix libenclave_quote_,nullquote))
 ifdef SGX
   crypto_wrappers += $(addsuffix .so,$(addprefix libcrypto_wrapper_,wolfcrypt-sgx))
   tls_wrappers += $(addsuffix .so,$(addprefix libtls_wrapper_,wolfssl-sgx))
-  enclave_quotes += $(addsuffix .so,$(addprefix libenclave_quote_,sgx-ecdsa))
+  enclave_quotes += $(addsuffix .so,$(addprefix libenclave_quote_,sgx-ecdsa)) \
+		    $(addsuffix .so,$(addprefix libenclave_quote_,sgx-la))
```
Comment (haosanzi) [inline enclave-tls/src/Makefile:31]: Please change to `enclave_quotes += $(addsuffix .so,$(addprefix libenclave_quote_,sgx-ecdsa sgx-la))`

File: enclave-tls/src/enclave_quotes/sgx-la/cleanup.c [hunk 2]
```diff
@@ -0,0 +1,15 @@
+#include <enclave-tls/log.h>
+#include <enclave-tls/enclave_quote.h>
+
```
Comment (haosanzi) [inline enclave-tls/src/enclave_quotes/sgx-la/cleanup.c:3]: Please delete this empty line

File: enclave-tls/src/enclave_quotes/sgx-la/collect_evidence.c [hunk 3]
```diff
@@ -0,0 +1,30 @@
+#include <enclave-tls/log.h>
+#include <enclave-tls/enclave_quote.h>
+#include <string.h>
+#include <sgx_report.h>
+
```
Comment (haosanzi) [inline enclave-tls/src/enclave_quotes/sgx-la/collect_evidence.c:5]: ditoo

File: enclave-tls/src/enclave_quotes/sgx-la/collect_evidence.c [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+#include "sgx_stub_u.h"
+#include "sgx_la.h"
+
+enclave_quote_err_t sgx_la_collect_evidence(enclave_quote_ctx_t *ctx,
+		attestation_evidence_t *evidence,
```
Comment (haosanzi) [inline enclave-tls/src/enclave_quotes/sgx-la/collect_evidence.c:10]: Please fix the format problem

File: enclave-tls/src/enclave_quotes/sgx-la/sgx_la.h [hunk 5]
```diff
@@ -0,0 +1,10 @@
+#ifndef _SGX_LA_H
+#define _SGX_LA_H
+
+#include "sgx_eid.h"
+
```
Comment (haosanzi) [inline enclave-tls/src/enclave_quotes/sgx-la/sgx_la.h:5]: Why include `sgx_eid.h`?

File: enclave-tls/src/enclave_quotes/sgx-la/ecalls.c [hunk 6]
```diff
@@ -0,0 +1,39 @@
+#include <string.h>
+#include <enclave-tls/enclave_quote.h>
+
+#include "sgx_report.h"
+#include "sgx_stub_t.h"
+#include "sgx_utils.h"
+
+sgx_status_t ecall_sgx_la_collect_evidence(attestation_evidence_t* evidence,
+				uint8_t* hash)
```
Comment (haosanzi) [inline enclave-tls/src/enclave_quotes/sgx-la/ecalls.c:9]: Please change to `*evidence`

File: enclave-tls/src/include/enclave-tls/err.h [hunk 7]
```diff
@@ -39,6 +39,9 @@
 /* The base of error code used by sgx-ecdsa */
 #define SGX_ECDSA_ERR_BASE		(0 << ERR_CODE_SUBCLASS_SHIFT)
 
+/* The base of error code used by sgx-la */
+#define	SGX_LA_ERR_BASE			(0 << ERR_CODE_SUBCLASS_SHIFT)
```
Comment (haosanzi) [inline enclave-tls/src/include/enclave-tls/err.h:43]: Please fix the format problem

File: enclave-tls/src/tls_wrappers/wolfssl-sgx/wolfssl_sgx.edl [hunk 8]
```diff
@@ -21,10 +23,14 @@ enclave {
...
 		public tls_wrapper_err_t ecall_wolfssl_cleanup([user_check] tls_wrapper_ctx_t *ctx);
+
+		/* This ecall function need be invoked by ocall_verify_certificate */
+		public sgx_status_t ecall_sgx_la_verify_report([in] sgx_report_t* report);
 	};
 
 	untrusted {
-		int ocall_verify_certificate([in, out, size=der_crt_len]uint8_t* der_crt, uint32_t der_crt_len);
+		int ocall_verify_certificate([in, out, size=der_crt_len]uint8_t* der_crt, uint32_t der_crt_len) 
+			allow(ecall_sgx_la_verify_report);
```
Comment (haosanzi) [inline enclave-tls/src/tls_wrappers/wolfssl-sgx/wolfssl_sgx.edl:33]: Please change to `[in, out, size=der_crt_len] uint8_t *der_crt`

File: enclave-tls/src/enclave_quotes/sgx-la/ecalls.c [hunk 9]
```diff
@@ -0,0 +1,38 @@
+#include <string.h>
+#include <enclave-tls/enclave_quote.h>
+#include "sgx_report.h"
+#include "sgx_stub_t.h"
+#include "sgx_utils.h"
+
+sgx_status_t ecall_sgx_la_collect_evidence(attestation_evidence_t *evidence,
+				uint8_t *hash)
```
Comment (jiazhang0) [inline enclave-tls/src/enclave_quotes/sgx-la/ecalls.c:8]: Please fix all indent issues.

File: enclave-tls/src/enclave_quotes/sgx-la/main.c [hunk 10]
```diff
@@ -0,0 +1,37 @@
+#include <stdio.h>
+#include <enclave-tls/enclave_quote.h>
+#include <enclave-tls/log.h>
+
+extern enclave_quote_err_t enclave_quote_register(enclave_quote_opts_t *);
+extern enclave_quote_err_t sgx_la_pre_init(void);
+extern enclave_quote_err_t sgx_la_init(enclave_quote_ctx_t *,
+				     enclave_tls_cert_algo_t algo);
```
Comment (jiazhang0) [inline enclave-tls/src/enclave_quotes/sgx-la/main.c:8]: ditto

File: enclave-tls/src/enclave_quotes/sgx-la/main.c [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+						 uint8_t *);
+extern enclave_quote_err_t sgx_la_verify_evidence(enclave_quote_ctx_t *,
+						attestation_evidence_t *, uint8_t *);
+extern enclave_quote_err_t sgx_la_cleanup(enclave_quote_ctx_t *);
+
+static enclave_quote_opts_t sgx_la_opts = {
+	.api_version = ENCLAVE_QUOTE_API_VERSION_DEFAULT,
+	.flags = ENCLAVE_QUOTE_FLAGS_DEFAULT,
+	.type = "sgx_la",
+	.priority = 53,
```
Comment (jiazhang0) [inline enclave-tls/src/enclave_quotes/sgx-la/main.c:21]: Plz change to 54.

50: sgx epid
51: tdx ecdsa
52: sgx ecdsa
53: tdx la
54: sgx la

**Review by jiazhang0 — APPROVED** [review]

**Pull request merged**